flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java

```diff
@@ -57,16 +57,56 @@
 /**
  * This class implements a top-level runner for {@link Task}s.
  */
-public final class FloRunner<T> {
+public class FloRunner<T> {
 
   private static final Logger LOG = LoggerFactory.getLogger(FloRunner.class);
+  private static final TerminationHookFactory DEFAULT_TERMINATION_HOOK_FACTORY = () ->
+      (Integer exitCode) -> {
+        LOG.info("Termination with exit code " + exitCode);
+        System.exit(exitCode);
+      };
 
   private final Logging logging = Logging.create(LOG);
   private final Collection<Closeable> closeables = new ArrayList<>();
   private final Config config;
+  private TerminationHookFactory terminationHookFactory;
+
+  public static class Builder {
+
+    private Config config = defaultConfig();
+    private TerminationHookFactory terminationHookFactory = DEFAULT_TERMINATION_HOOK_FACTORY;
+
+    public Builder setConfig(Config config) {
+      this.config = config;
+      return this;
+    }
+
+    public Builder setTerminationHookFactory(TerminationHookFactory terminationHookFactory) {
+      this.terminationHookFactory = terminationHookFactory;
+      return this;
+    }
+
+    public FloRunner build() {
+      return new FloRunner(this);
+    }
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static FloRunner createDefault() {
+    return newBuilder().build();
+  }
 
   private FloRunner(Config config) {
```

Comment:
We have the opportunity to break backward compatibility, so let's break it badly.